pkg/ingester/errors.go

```diff
@@ -75,6 +79,17 @@ func wrapWithUser(err error, userID string) error {
 	return fmt.Errorf("user=%s: %w", userID, err)
 }
 
+// wrapOrAnnotateWithUser checks whether the given error is safe, and in that case
+// wraps it with wrapWithUser. Otherwise, the error annotated with annotateWithUser
+// is returned.
+func wrapOrAnnotateWithUser(err error, userID string) error {
+	var safe safeToWrap
+	if errors.As(err, &safe) {
+		return wrapWithUser(err, userID)
```

Comment:
Can we remove `wrapWithUser` and `annotateWithUser` to make sure people always use the method that ensures it's safe to do so?

Reply:
I have removed them.